src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -614,166 +676,199 @@ mod integration_tests {
 
-            let pump = sink.send_all(events);
-            let (sink, _) = rt.block_on(pump).unwrap();
-            // drop the sink so it closes all its connections
-            drop(sink);
+        let pump = sink.send_all(events);
+        let (sink, _) = rt.block_on(pump).unwrap();
+        // drop the sink so it closes all its connections
+        drop(sink);
 
-            let mut request = GetLogEventsRequest::default();
-            request.log_stream_name = stream_name.clone().into();
-            request.log_group_name = GROUP_NAME.into();
-            request.start_time = Some(timestamp.timestamp_millis());
+        let mut request = GetLogEventsRequest::default();
+        request.log_stream_name = stream_name.clone().into();
+        request.log_group_name = GROUP_NAME.into();
+        request.start_time = Some(timestamp.timestamp_millis());
 
-            let client = CloudWatchLogsClient::new(region);
+        let client = create_client(region).unwrap();
 
-            let response = rt.block_on(client.get_log_events(request)).unwrap();
+        let response = rt.block_on(client.get_log_events(request)).unwrap();
 
-            let events = response.events.unwrap();
+        let events = response.events.unwrap();
 
-            let output_lines = events
-                .into_iter()
-                .map(|e| e.message.unwrap())
-                .collect::<Vec<_>>();
+        let output_lines = events
+            .into_iter()
+            .map(|e| e.message.unwrap())
+            .collect::<Vec<_>>();
 
-            assert_eq!(output_lines, input_lines);
-        }
+        assert_eq!(output_lines, input_lines);
+    }
 
-        // Run multi partition test
-        {
-            let stream_name = gen_stream();
+    #[test]
+    fn cloudwatch_insert_log_event_batched() {
+        let mut rt = Runtime::new().unwrap();
 
-            let region = Region::Custom {
-                name: "localstack".into(),
-                endpoint: "http://localhost:6000".into(),
-            };
+        let stream_name = gen_stream();
 
-            let client = CloudWatchLogsClient::new(region.clone());
-            ensure_group(region);
+        let region = Region::Custom {
+            name: "localstack".into(),
+            endpoint: "http://localhost:6000".into(),
+        };
+        ensure_group(region.clone());
 
-            let config = CloudwatchLogsSinkConfig {
-                group_name: GROUP_NAME.into(),
-                stream_name: format!("{}-{{{{key}}}}", stream_name).into(),
-                region: RegionOrEndpoint::with_endpoint("http://localhost:6000".into()),
-                ..Default::default()
-            };
+        let config = CloudwatchLogsSinkConfig {
+            stream_name: stream_name.clone().into(),
+            group_name: GROUP_NAME.into(),
+            region: RegionOrEndpoint::with_endpoint("http://localhost:6000".into()),
+            batch_size: Some(2),
```

Comment:
Is this the only difference from the previous test case?

Reply:
Yes, I wanted to ensure all partitions were being driven.